src/AerospikeCache.php

```diff
@@ -0,0 +1,135 @@
+    private $namespace;
+
+    /** @var string */
+    private $set;
+
+    public function __construct(
+        \Aerospike $aerospike,
+        string $namespace,
+        string $set,
+        string $cacheNamespace = '',
+        int $defaultLifetime = 0
+    ) {
+        $this->aerospike = $aerospike;
+        $this->namespace = $namespace;
+        $this->set = $set;
+        parent::__construct($cacheNamespace, $defaultLifetime);
+    }
+
+    protected function doFetch(array $ids): array
+    {
+        $result = [];
+
+        $keys = $this->initializeKeysForAerospike($ids);
+
+        $this->aerospike->getMany($keys, $records);
+
+        foreach ($records as $record) {
+            if ($record['metadata'] !== null) {
+                $result[$record['key']['key']] = $record['bins'][self::WRAPPER_NAME] ?? null;
+            }
+        }
+
+        return $result;
+    }
+
+    private function initializeKeysForAerospike(array $ids): array
+    {
+        return array_map(
+            function ($id) {
+                return $this->createKey($id);
+            },
+            $ids
+        );
+    }
+
+    protected function doHave($id)
+    {
+        return $this->aerospike->get($this->createKey($id), $ignoredRecord) === \Aerospike::OK;
+    }
+
+    protected function doClear($namespace = ''): bool
+    {
+        if ($namespace === '') {
+            $statusCode = $this->aerospike->truncate($this->namespace, $this->set, 0);
+            $cleared = $this->isStatusOkOrNotFound($statusCode);
+        } else {
+            $removedAllRecords = true;
+            $clearNamespace = function ($record) use ($namespace, &$removedAllRecords): void {
+                if (mb_strpos($record['key']['key'], $namespace) === 0) {
+                    $statusCodeFromRemove = $this->aerospike->remove($record['key']);
+
+                    if (!$this->isStatusOkOrNotFound($statusCodeFromRemove)) {
+                        $removedAllRecords = false;
+                    }
+                }
+            };
+
+            $statusCodeFromScan = $this->aerospike->scan($this->namespace, $this->set, $clearNamespace);
+            $cleared = $removedAllRecords && $this->isStatusOkOrNotFound($statusCodeFromScan);
+        }
+
+        return $cleared;
+    }
+
+    protected function doDelete(array $ids): bool
+    {
+        $removedAllItems = true;
+
+        foreach ($ids as $id) {
+            $statusCode = $this->aerospike->remove($this->createKey($id));
+            if (!$this->isStatusOkOrNotFound($statusCode)) {
+                $removedAllItems = false;
+            }
+        }
+
+        return $removedAllItems;
+    }
+
+    /**
+     * @param int $lifetime
+     * @return string[] keys of values that failed during save operation
+     */
+    protected function doSave(array $values, $lifetime): array
+    {
+        $failed = [];
+        foreach ($values as $key => $value) {
+            $data = [self::WRAPPER_NAME => $value];
```

Comment:
I'm also curious about this `wrapper`
- why is even here?
- if I change the code to following code (remove the wrapping and pass `$value` directly)
```php
$statusCode = $this->aerospike->put(
    $this->createKey($key),
    $value,
    $lifetime,
    [\Aerospike::OPT_POLICY_KEY => \Aerospike::POLICY_KEY_SEND]
);
```
- all tests pass 🤔

Reply:
I think i understand. It is a behaviour that should not be changed and it might pass in some scenarios with others it will fail on the side of aerospike. Would something like this do? 
`

        $aerospikeMock = $this->createMock(\Aerospike::class);

        $aerospikeCache = new AerospikeCache($aerospikeMock, 'test', 'cache', 'testNamespace');
        $reflection = new \ReflectionClass(AerospikeCache::class);
        $doSaveMethod = $reflection->getMethod('doSave');
        $doSaveMethod->setAccessible(true);

        $testValue = ['testKey' => 'testArray'];
        $testKey = ['ns' => 'aerospike', 'set' => 'cache', 'key' => 'testKey'];
        $aerospikeMock->method('initkey')->willReturn($testKey);
        $aerospikeMock->expects($this->once())->method('put')-> with(
            $testKey,
            ['data' => $testValue['testKey']],
            0,
            [\Aerospike::OPT_POLICY_KEY => \Aerospike::POLICY_KEY_SEND]
        );
        $doSaveMethod->invokeArgs($aerospikeCache, [$testValue, 0]);`

Reply:
Test is done. But what about the `wrapper`, why is it there?

Reply:
In aerospike the records are stored using Bins. Bin is like a field. So you could have record, with various number of  fields. We use only ona Bin and that is data. 
The thing is if you would try to save directly for example array 
`$valueToSave = ['helloWorld'] ` 
The save would fail. You need string keys. Like this the value is stored in single Bin (serialized) without any issues with the keys.